src/casts/specify-representation.rkt

```diff
@@ -411,24 +388,20 @@ but a static single assignment is implicitly maintained.
     (define cast-tuple-label (Code-Label cast-tuple))
     (define cast-label (Code-Label cast))
     (define cast-tuple-c : D0-Code
-      (code$ (tpl-val t1 t2 l i count)
-        (If (op$ = i count)
-            (op$ Alloc count)
-            (begin$ (assign$ val (op$ Array-ref tpl-val i))
-                    (assign$ t1a
-                      (sr-tagged-array-ref
-                       t1 TYPE-TUPLE-TAG (sr-plus TYPE-TUPLE-ELEMENTS-OFFSET i)))
-                    (assign$ t2a
-                      (sr-tagged-array-ref
-                       t2 TYPE-TUPLE-TAG (sr-plus TYPE-TUPLE-ELEMENTS-OFFSET i)))
-                    (assign$ casted-val
-                      (App-Code cast-label `(,val ,t1a ,t2a ,l ,ZERO-IMDT)))
-                    (assign$ tmp-crcn
-                      (App-Code
-                       cast-tuple-label
-                       `(,tpl-val ,t1 ,t2 ,l ,(sr-plus (Quote 1) i) ,count)))
-                    (op$ Array-set! tmp-crcn i casted-val)
-                    tmp-crcn))))
+      (code$ (val t1 t2 l)
+        (assign$ count
+          (sr-tagged-array-ref t2 TYPE-TUPLE-TAG TYPE-TUPLE-COUNT-INDEX))
+        (assign$ new-val (op$ Alloc count))
+        (assign$ iters (op$ + TYPE-TUPLE-ELEMENTS-OFFSET count))
+        (repeat$ (i TYPE-TUPLE-ELEMENTS-OFFSET iters) (_ UNIT-IMDT)
+          (begin$
+            (assign$ val-i (op$ - i TYPE-TUPLE-ELEMENTS-OFFSET))
+            (assign$ vala (op$ Array-ref val val-i))
+            (assign$ t1a (sr-tagged-array-ref t1 TYPE-TUPLE-TAG i))
+            (assign$ t2a (sr-tagged-array-ref t2 TYPE-TUPLE-TAG i))
+            (assign$ casted-vala (app-code$ cast-label vala t1a t2a l ZERO-IMDT))
+            (op$ Array-set! new-val val-i casted-vala)))
```

Comment:
if repeat$ doesn't have an implicit begin block we should add one.

Reply:
It does.

Reply:
Then we should use it here.

Reply:
I completely missed that begin for some reason.

Reply:
Done.